webdriver/tests/bidi/network/remove_intercept/remove_intercept.py

```diff
@@ -0,0 +1,90 @@
+# META: timeout=long
+
+import asyncio
+
+import pytest
+from webdriver.bidi.modules.script import ScriptEvaluateResultException
+
+from .. import (
+    assert_before_request_sent_event,
+    assert_response_event,
+)
+
+PAGE_EMPTY_HTML = "/webdriver/tests/bidi/network/support/empty.html"
+PAGE_EMPTY_TEXT = "/webdriver/tests/bidi/network/support/empty.txt"
+PAGE_OTHER_TEXT = "/webdriver/tests/bidi/network/support/other.txt"
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("phase", ["beforeRequestSent", "responseStarted"])
+async def test_remove_intercept(
+    bidi_session, wait_for_event, url, setup_network_test, fetch, phase
+):
+    network_events = await setup_network_test(
+        events=[
+            "network.beforeRequestSent",
+            "network.responseStarted",
+            "network.responseCompleted",
+        ]
+    )
+    before_request_sent_events = network_events["network.beforeRequestSent"]
+    response_started_events = network_events["network.responseStarted"]
+    response_completed_events = network_events["network.responseCompleted"]
+
+    text_url = url(PAGE_EMPTY_TEXT)
+    intercept = await bidi_session.network.add_intercept(
+        phases=[phase],
+        url_patterns=[{"type": "string", "pattern": text_url}],
+    )
+
+    assert isinstance(intercept, str)
+
+    on_network_event = wait_for_event(f"network.{phase}")
+
+    # Request to top_context should be blocked and throw a ScriptEvaluateResultException
+    # from the AbortController.
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await fetch(text_url)
+
+    await on_network_event
+
+    if phase == "beforeRequestSent":
+        assert len(before_request_sent_events) == 1
+        assert_before_request_sent_event(
+            before_request_sent_events[0], is_blocked=True, intercepts=[intercept]
+        )
+        assert len(response_started_events) == 0
+    elif phase == "responseStarted":
+        assert len(before_request_sent_events) == 1
```

Comment:
Optionally move L52 and L58 to outside the `if`, since they perform the same assertion in both flows.

Reply:
Sure!